tests/security/pam/pam_u2f.pm

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: Update pam_u2f to 1.1.1 (or later)
+#          Add support to FIDO2 (move from libu2f-host+libu2f-server to libfido2)
+#          Add support to User Verification
+#          Add support to PIN Verification
+#          Add support to Resident Credentials
+#          Add support to SSH credential format
+#          This adds support for FIDO2 (yubikey or other)
+#          with pam and provides additional verifications (User, PIN)
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#104181 tc#1769990
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use base 'consoletest';
+use utils 'zypper_call';
+
+sub run {
+    select_console('root-console');
+    zypper_call('in pam_u2f');
+
+    # Package version check
+    my $target_ver = '1.1.1';
+    my $current_ver = script_output q(pamu2fcfg --version | awk '{print $2}');
+    record_info("Current pam_u2f version is $current_ver, tarrget version is $target_ver");
+    die 'The package version is lower than expected' if ($current_ver lt $target_ver);
+
+    # Package change log check
+    my $change_log = script_output('rpm -q pam_u2f --changelog');
+    if ($change_log =~ m/Add support to FIDO2/
+        && $change_log =~ m/Add support to User Verification/
+        && $change_log =~ m/Add support to PIN Verification/
+        && $change_log =~ m/Add support to Resident Credentials/
+        && $change_log =~ m/Add support to SSH credential format/)
+    {
+        record_info('All required patches are set');
+    }
+    else {
+        die('Not all required patches are set, please check with developer');
+    }
+
+    # 'pamu2fcfg' command test, we don't have available yubikey
+    # However, we can still check this command can work
+    script_run('pamu2fcfg > /tmp/pam_u2f.log 2>&1');
+    assert_script_run('cat /tmp/pam_u2f.log | grep "No device found. Aborting."');
```

Comment:
I wonder if there could be the possibility to use a qemu virtual device for emulating the u2f token, I know about this, https://github.com/MattGorko/libu2f-emu

(It's just  a random thought, don't take me seriously, other than the actual test needs to also be done manually to really verify that the support is working)

Reply:
It is a GREAT share, let me try, Thanks much @foursixnine

Reply:
Seems the code can't work perfectly on my setup:
```
susetest:~/libu2f-emu # ./configure 
checking for doxygen... doxygen
checking for dot... dot
./configure: line 5044: AX_COMPILER_FLAGS: command not found
./configure: line 5045: syntax error near unexpected token `-std=c99,'
./configure: line 5045: `AX_CHECK_COMPILE_FLAG(-std=c99,'

```
I will try to find a yubikey to complete this feature task. but for current codes can check the new package version is there